module/os/linux/zfs/vdev_object_store.c

```
@@ -1254,6 +1255,18 @@ agent_reader(void *arg)
 		vos->vos_serial_done[VOS_SERIAL_CREATE_POOL] = B_TRUE;
 		cv_broadcast(&vos->vos_outstanding_cv);
 		mutex_exit(&vos->vos_outstanding_lock);
+	} else if (strcmp(type, AGENT_TYPE_CREATE_POOL_FAILED) == 0) {
```

I was curious about returning the error as a new type (as you've done here) vs the same type (AGENT_TYPE_CREATE_POOL_DONE) with a field in the nvlist indicating success/failure.  e.g. "error" -> <string> vs there's no "error" nvpair.  The protocol might be easier to understand if every request had a response with "Type" -> <request name> "done".

---

I was following the "open pool", "pool open done", "pool open failed" model. :-D
But we could just return an error code as you suggest. If we do that, we should also fix the "open pool done" API to do the same.

I am happy to make the change.

I assume George was the author of the "open pool" API. @grwilson do you want to weigh in?